serverless/src/generateNotebook/handler.js

```diff
@@ -0,0 +1,196 @@
+
+let s3Client
+
+/**
+ * Generates Jupyter notebook based on parameters.
+ * It fetches granule information from Graphql, creates a notebook template,
+ * and uploads the result to S3, providing a signed URL for download.
+ *
+ * @param {Object} event Details about the HTTP request that it received
+ */
+const generateNotebook = async (event) => {
+  const { body } = event
+
+  const { defaultResponseHeaders } = getApplicationConfig()
+
+  const { GENERATE_NOTEBOOKS_BUCKET_NAME: generateNotebooksBucketName } = process.env
+
+  if (s3Client == null) {
+    s3Client = getS3Client()
+  }
+
+  const params = JSON.parse(body)
+
+  const {
+    boundingBox,
+    referrerUrl,
+    granuleId,
+    variableId
+  } = params
+
+  let boundingBoxValues
+  if (boundingBox) {
+    const [minLon, minLat, maxLon, maxLat] = boundingBox.split(', ')
+    boundingBoxValues = {
+      minLon,
+      minLat,
+      maxLon,
+      maxLat
+    }
+  }
+
+  const { headers } = event
+
+  const earthdataEnvironment = determineEarthdataEnvironment(headers)
+
+  const jwtToken = getJwtToken(event)
+
+  const graphQLHeader = {
+    'Content-Type': 'application/json'
+  }
+
+  // The authorizer will have already validated the token if it exists
+  if (jwtToken) {
+    const { access_token: accessToken } = await
+    getAccessTokenFromJwtToken(jwtToken, earthdataEnvironment)
+    graphQLHeader.Authorization = `Bearer ${accessToken}`
+  }
+
+  const { graphQlHost } = getEarthdataConfig(earthdataEnvironment)
+
+  const graphQlUrl = `${graphQlHost}/api`
+
+  const graphqlQuery = `
+    query Granules($granuleParams: GranulesInput, $variablesParams: VariablesInput) {
+      granules(params: $granuleParams) {
+        count
+        items {
+          conceptId
+          title
+          collection {
+            conceptId
+            title
+            variables(params: $variablesParams) {
+              items {
+                name
+              }
+            }
+          }
+        }
+      }
+    }
+  `
+
+  try {
+    const granuleResponse = await axios({
+      url: graphQlUrl,
+      method: 'post',
+      data: {
+        query: graphqlQuery,
+        variables: {
+          granuleParams: {
+            conceptId: granuleId
+          },
+          variablesParams: {
+            conceptId: variableId
+          }
+        }
+      },
+      headers: graphQLHeader
+    })
+
+    const { data: responseData } = granuleResponse
+    const { data } = responseData
+    const { granules } = data
+    const { items: granulesItems } = granules
+    const { collection, title: granuleTitle } = granulesItems[0]
+
+    const { conceptId: collectionId, title: collectionTitle, variables } = collection
+
+    const { items: variableItems } = variables
+    const { name: variableName } = variableItems[0]
+
+    // Read the Jupyter notebook template file
+    const templateContent = fs.readFileSync('serverless/src/generateNotebook/notebookTemplate.ipynb', 'utf-8')
```

Comment:
Can we import this file like we would a JSON file? So we wouldn't have to use `fs.readFileSync` and just have esbuild include it in the build?